Mathlib/Probability/Moments/ComplexMGF.lean

```diff
@@ -0,0 +1,315 @@
+  · refine ae_of_all _ fun ω ε hε ↦ ?_
+    simp only [norm_mul, norm_pow, norm_real, Real.norm_eq_abs, Complex.norm_eq_abs]
+    rw [Complex.abs_ofReal, Complex.abs_exp]
+    simp only [mul_re, ofReal_re, ofReal_im, mul_zero, sub_zero]
+    gcongr
+    have : ε = z + (ε - z) := by simp
+    rw [this, add_re, add_mul]
+    gcongr _ + ?_
+    refine (le_abs_self _).trans ?_
+    rw [abs_mul]
+    gcongr
+    refine (abs_re_le_abs _).trans ?_
+    simp only [Metric.mem_ball, dist_eq_norm, Complex.norm_eq_abs] at hε
+    exact hε.le
+  · refine integrable_pow_abs_mul_exp_add_of_integrable_exp_mul ?_ ?_ ?_ ?_ (t := t) (n + 1)
+    · exact h_subset (add_half_inf_sub_mem_Ioo hlu)
+    · exact h_subset (sub_half_inf_sub_mem_Ioo hlu)
+    · positivity
+    · exact lt_of_lt_of_le (by simp [ht]) (le_abs_self _)
+  · refine ae_of_all _ fun ω ε hε ↦ ?_
+    simp only
+    simp_rw [pow_succ, mul_assoc]
+    refine HasDerivAt.const_mul _ ?_
+    simp_rw [← smul_eq_mul, Complex.exp_eq_exp_ℂ]
+    convert hasDerivAt_exp_smul_const (X ω : ℂ) ε using 1
+    rw [smul_eq_mul, mul_comm]
+
+/-- For all `z : ℂ` with `z.re ∈ interior (integrableExpSet X μ)`,
+`complexMGF X μ` is differentiable at `z` with derivative `μ[X * exp (z * X)]`. -/
+theorem hasDerivAt_complexMGF (hz : z.re ∈ interior (integrableExpSet X μ)) :
+    HasDerivAt (complexMGF X μ) μ[fun ω ↦ X ω * cexp (z * X ω)] z := by
+  convert hasDerivAt_integral_pow_mul_exp hz 0
+  · simp [complexMGF]
+  · simp
+
+/-- `complexMGF X μ` is holomorphic on the vertical strip
+`{z | z.re ∈ interior (integrableExpSet X μ)}`. -/
+theorem differentiableOn_complexMGF :
+    DifferentiableOn ℂ (complexMGF X μ) {z | z.re ∈ interior (integrableExpSet X μ)} := by
+  intro z hz
+  have h := hasDerivAt_complexMGF hz
+  rw [hasDerivAt_iff_hasFDerivAt] at h
+  exact h.hasFDerivWithinAt.differentiableWithinAt
+
+theorem analyticOnNhd_complexMGF :
+    AnalyticOnNhd ℂ (complexMGF X μ) {z | z.re ∈ interior (integrableExpSet X μ)} :=
+  differentiableOn_complexMGF.analyticOnNhd (isOpen_interior.preimage Complex.continuous_re)
+
+/-- `complexMGF X μ` is analytic on the vertical strip
+  `{z | z.re ∈ interior (integrableExpSet X μ)}`. -/
+theorem analyticOn_complexMGF :
+    AnalyticOn ℂ (complexMGF X μ) {z | z.re ∈ interior (integrableExpSet X μ)} :=
+  analyticOnNhd_complexMGF.analyticOn
+
+/-- `complexMGF X μ` is analytic at any point `z` with `z.re ∈ interior (integrableExpSet X μ)`. -/
+lemma analyticAt_complexMGF (hz : z.re ∈ interior (integrableExpSet X μ)) :
+    AnalyticAt ℂ (complexMGF X μ) z :=
+  analyticOnNhd_complexMGF z hz
+
+end Analytic
+
+section Deriv
+
+/-! ### Iterated derivatives of `complexMGF` -/
+
+/-- For `t : ℝ` with `t ∈ interior (integrableExpSet X μ)`, the derivative of the function
+`x ↦ μ[X ^ n * rexp (x * X)]` at `t` is `μ[X ^ (n + 1) * rexp (t * X)]`. -/
+lemma hasDerivAt_integral_pow_mul_exp_real (ht : t ∈ interior (integrableExpSet X μ)) (n : ℕ) :
```

Comment:
Maybe move this one to the file on `MGFAnalytic`? It is purely on the real MGF, if I understand correctly.